drivers/auxiliary/myDewControllerPro.cpp

```diff
@@ -0,0 +1,945 @@
+    }
+    return true;
+
+}
+
+bool myDewControllerPro::setChoice(int testInt, const char *positiveChoice, const char *negativeChoice, const char *errMessage)
+{
+    const char* mask = testInt == 1 ? positiveChoice : negativeChoice;
+    if (!sendCommand(mask, nullptr)) {
+        LOG_INFO(errMessage);
+
+        return false;
+    }
+    return true;
+}
+
+
+
+bool myDewControllerPro::setTempCalibrations(float ch1, float ch2, float ch3, int ambient)
+{
+    char cmd[MDCP_CMD_LEN + 1];
+
+
+    snprintf(cmd, MDCP_CMD_LEN + 1, MDCP_SET_TEMP_CH1_OFFSET, ch1);
+    if (!sendCommand(cmd, nullptr)) {
+        LOG_INFO("Failed to set CH1 offset");
+        LOG_INFO(cmd);
+        return false;
+    }
+    snprintf(cmd, MDCP_CMD_LEN + 1, MDCP_SET_TEMP_CH2_OFFSET, ch2);
+    if (!sendCommand(cmd, nullptr)) {
+        LOG_INFO("Failed to set CH2 offset");
+        LOG_INFO(cmd);
+        return false;
+    }
+    snprintf(cmd, MDCP_CMD_LEN + 1, MDCP_SET_TEMP_CH3_OFFSET, ch3);
+    if (!sendCommand(cmd, nullptr)) {
+        LOG_INFO("Failed to set CH3 offset");
+        LOG_INFO(cmd);
+        return false;
+    }
+
+    snprintf(cmd, MDCP_CMD_LEN + 1, MDCP_SET_AMB_TEMP_OFFSET, ambient);
+    if (!sendCommand(cmd, nullptr)) {
+        LOG_INFO("Failed to set CH3 offset");
+        LOG_INFO(cmd);
+        return false;
+    }
+    return true;
+
+
+
+}
+
+bool myDewControllerPro::setFanTempTrigger(int tempOn, int tempOff)
+{
+    char cmd[MDCP_CMD_LEN + 1];
+
+
+    snprintf(cmd, MDCP_CMD_LEN + 1, MDCP_SET_FAN_ON_TEMP, tempOn);
+    if (!sendCommand(cmd, nullptr)) {
+        LOG_INFO("Failed to set temp on");
+        LOG_INFO(cmd);
+        return false;
+    }
+    snprintf(cmd, MDCP_CMD_LEN + 1, MDCP_SET_FAN_OFF_TEMP, tempOff);
+    if (!sendCommand(cmd, nullptr)) {
+        LOG_INFO("Failed to set CH2 offset");
+        LOG_INFO(cmd);
+        return false;
+    }
+
+    return true;
+
+
+
+}
+
+bool myDewControllerPro::zeroTempCalibrations() {
+
+    if (!sendCommand(MDCP_CLEAR_TEMP_OFFSETS, nullptr)) {
+        LOG_INFO("Failed to zero temp offset");
+
+        return false;
+    }
+    if (!sendCommand("e0#", nullptr)) {
+        LOG_INFO("Failed to zero ambtemp offset");
+
+        return false;
+    }
+    return true;
+
+
+}
+
+
+
+
+
+
+
+
+bool myDewControllerPro::ISNewSwitch(const char *dev, const char *name, ISState *states, char *names[], int n)
+{
+    if (!dev || strcmp(dev, getDeviceName()))
+        return false;
+
+    if (!strcmp(name, CH1CH2BoostSP.name))
+    {
+        IUUpdateSwitch(&CH1CH2BoostSP, states, names, n);
+        CH1CH2BoostSP.s = IPS_BUSY;
+        cancelOutputBoost();
+        if (CH1CH2BoostS[CH1_BOOST_100].s == ISS_ON) {
+            setOutputBoost(CH1_BOOST_100);
+        }
+        if (CH1CH2BoostS[CH2_BOOST_100].s == ISS_ON) {
+            setOutputBoost(CH2_BOOST_100);
+        }
+        CH1CH2BoostSP.s = IPS_OK;
+        IDSetSwitch(&CH1CH2BoostSP, nullptr);
+        readMainValues();
+        return true;
```

Comment:
Example below on how to change this to modern INDI convention. Please apply it to the rest of the code.

CH1CH2BoostSP.update(states, names, n);
CH1CH2BoostSP.setState(IPS_BUSY);
....
if (CH1CH2BoostSP[CH1_BOOST_100].getState() == ISS_ON) {
.....
CH1CH2BoostSP.setState(IPS_OK);
CH1CH2BoostSP.apply();

Reply:
No problem, I know how to do that as I started that way and changed all my code (opps) as I thought it was wrong.